tiledb/api/src/query/mod.rs

```diff
@@ -74,12 +78,30 @@ impl<'ctx> QueryBase<'ctx> {
         })
         .map(|_| c_status)
     }
+
+    pub fn array(&self) -> &Array<'ctx> {
+        /*
+         * TODO: if this is pub then it defeats the purpose of the query API consuming
```

Comment:
Probably delete this comment, the main reason to consume the array is to prevent multiple queries running simultaneously on the same open array, and this `pub fn` will not enable that.

Reply:
Took me a second, but I'm pretty sure you mean that since its returning `&Array<'ctx>`, we can't create a query from that hence no worries about this being pub since it allows read access to the array without exposing the non-Sync'ness of our underlying Array type?

Reply:
Right, that is what I mean.